### ISSUE TITLE
time sent: display date if deviating

### ISSUE THREAD

**Issue by testbird** [deltachat/deltachat-core#151]
The "time sent" is now displayed correctly (https://github.com/deltachat/deltachat-core/commit/9f4ab2fd876b623a23666522d095cfa3da126664).

But when some messages are delayed or only downloaded after some offline-days, they look as if they where from the current day (and even right out of the future, if sent later in the day).

It seems ok to display the messages in the order as arrived but confusing situations could be avoided, if the true date (day) could also be displayed in case it is different than the current date (as indicated by the markers in message list).

**Comment by r10s** [comment]
took me a moment to understand.

so you want to display _day/month/year_ beside the time if the arrival time is on another day as the sending time. yes, this sounds reasonable, however, as an workaround, currently, the whole times for such cornercases can be viewed at "info".

**Comment by testbird** [comment]
Yes, it's clearer, the appearing as if from the current day can lead to funny confusions. Or make senders look strangely repetitive or inadequate. ;-)

**Comment by r10s** [comment]
i agree, this should be optimized sooner or later.

**Comment by r10s** [comment]
this is an android-issue, there's nothing the core can do about this.

however, iirc, the upcoming android version from https://github.com/deltachat/deltachat-android-ii/ will have the date visible in a non-scrolling area permanently (as signal or newer versions of telegram - don't know whatsapp)

**Comment by testbird** [comment]
Why can't the core add the date at which a messages was sent (e.g. Aug. 20) to the message, if it is a date from the past (today is Aug. 28)?

That would make it visible that the message got delayed, is old, sent in the past.

**Comment by r10s** [comment]
the core [always returns full date+time](https://deltachat.github.io/api/classdc__msg__t.html#af667c538fd07771eb94d6c5d1879b906), it's the UI that formats the timestamp and decides eg. only the time, sth. like "minutes ago" or whatever.

**Comment by testbird** [comment]
Ah ok,

But if moving issues is not possible, all the issues should only be closed if actually solved in the new frontend.

**Comment by testbird** [comment]
Actually, the core could insert [Delayed since: Aug. 20th] to make it known.
The new date visualization won't help if it only shows the received at date (message sorting).

**Comment by r10s** [comment]
yes, letting the core add sth. like `[Message from: Aug. 20th]` is probably a good idea. re-opening :)
